integrations/gradle-bloop/src/main/scala/bloop/integrations/gradle/model/BloopConverter.scala

```diff
@@ -0,0 +1,247 @@
+  ): Config.Scala = {
+    val compilerName = parameters.compilerName
+    val scalaCompilerArtifact = artifacts
+      .find(_.getName == compilerName)
+      .getOrElse(throw new GradleException(s"${compilerName} is not added as dependency"))
+    val scalaVersion = scalaCompilerArtifact.getModuleVersion.getId.getVersion
+    val scalaGroup = scalaCompilerArtifact.getModuleVersion.getId.getGroup
+
+    val scalaCompileTaskName = sourceSet.getCompileTaskName("scala")
+    val scalaCompileTask = project.getTask[ScalaCompile](scalaCompileTaskName)
+    val compilerClasspath = scalaCompileTask.getScalaClasspath.asScala.map(_.toPath).toArray
+
+    val opts = scalaCompileTask.getScalaCompileOptions
+    val compilerOptions = optionSet(opts).toArray
+
+    Config.Scala(
+      scalaGroup,
+      parameters.compilerName,
+      scalaVersion,
+      compilerOptions,
+      compilerClasspath
+    )
+  }
+
+  private def getJavaConfig(project: Project, sourceSet: SourceSet): Config.Java = {
+    val javaCompileTaskName = sourceSet.getCompileTaskName("java")
+    val javaCompileTask = project.getTask[JavaCompile](javaCompileTaskName)
+    val opts = javaCompileTask.getOptions
+
+    val specs = new DefaultJavaCompileSpec()
+    specs.setCompileOptions(opts)
+
+    val builder = new JavaCompilerArgumentsBuilder(specs)
+      .includeMainOptions(true)
+      .includeClasspath(false)
+      .includeSourceFiles(false)
+      .includeLauncherOptions(false)
+    val args = builder.build().asScala.toArray
+
+    Config.Java(args)
+  }
+
+  private def ifEnabled[T](option: Boolean)(value: T): Option[T] =
+    if (option) Some(value) else None
+
+  private def optionSet(options: ScalaCompileOptions): Set[String] = {
+    // based on ZincScalaCompilerArgumentsGenerator
+    val baseOptions: Set[String] = Seq(
+      ifEnabled(options.isDeprecation)("-deprecation"),
+      ifEnabled(options.isUnchecked)("-unchecked"),
+      ifEnabled(options.isOptimize)("-optimize"),
+      ifEnabled(options.getDebugLevel == "verbose")("-verbose"),
+      ifEnabled(options.getDebugLevel == "debug")("-Ydebug"),
+      Option(options.getEncoding).map(encoding => s"--encoding $encoding"),
+      Option(options.getDebugLevel).map(level => s"-g:$level")
+    ).flatten.toSet
+
+    val loggingPhases: Set[String] =
+      Option(options.getLoggingPhases)
+        .map(_.asScala.toSet)
+        .getOrElse(Set.empty)
+        .map(phase => s"-Ylog:$phase")
+
+    val additionalOptions: Set[String] =
+      options.getAdditionalParameters.asScala.toSet
+
+    baseOptions.union(loggingPhases).union(additionalOptions)
+  }
+
+  private val scalaCheckFramework = Config.TestFramework(
+    List(
+      "org.scalacheck.ScalaCheckFramework"
+    ))
+
+  private val scalaTestFramework = Config.TestFramework(
+    List(
+      "org.scalatest.tools.Framework",
+      "org.scalatest.tools.ScalaTestFramework"
+    )
+  )
+
+  private val specsFramework = Config.TestFramework(
```

Comment:
What do you think of sharing these default frameworks with `MojoImplementation` by adding them to the `jsonConfig` module? The idea would be to have a value called `DefaultTestFrameworks` that can be invoked by both integrations.
  
Another thing that we better do now here is to have a function `def detectTestFrameworks(artifacts: List[Config.Artifact]): List[Config.TestFramework]` that would add these only if the classpath contains the artifact associated with every framework (scalatest, specs, junit, etc).
  
(This last point is minor, if you don't feel like taking on it we can add a ticket so that another contributor does it :smile:)

Reply:
Good idea, will do it